src/manual/devel/best-practices/microservices.md

```diff
@@ -0,0 +1,307 @@
+---
+title: Building Microservices
+order: 20
+---
+
+The TaskCluster microservices are all maintained independently, but we share responsibility for maintaining them.
+This shared responsibility is easier for everyone if the implementations are similar, avoiding surprises when moving from one service to another.
+This document aims to collect the practices and standards we've agreed on.
+
+Most services do not live up to these standards -- TaskCluster is a work in progress -- but pull requests to bring services up-to-date with one or more points are always appreciated.
+Many of these aren't hard-and-fast rules, either.
+If there's good reason to do otherwise, do so!
+
+## Package Mechanics
+
+### Repository
+
+Services should be in a Github repository in the `taskcluster` organization, with the repo name having the prefix `taskcluster-`.
+
+### Source Layout
+
+Include all source in the `src/` directory, and all tests in `test/`.
+The babel-compile configuration will compile those to `lib/` and `.test/`, respectively.
+Within the `src` directory, the main script should be named `main.js`.
```

Comment:
We should mention what the `main.js` script is here as well, and encourage it's use.  Either referencing the section later, or adding something:

The `src` directory should have a `main.js` file which contains a loader using the `taskcluster-lib-loader`.  This file will be used to initialise components and should serve as the main entry point to the service.  This file will be transpiled to `lib/main.js` and should be set as the 'main' key in the package.json file.